[Lim Jun Hao] iP

JunHao247 opened the pull request:
# **JamesBot**
>“Your mind is for having ideas, not holding them.” – David Allen [(source)](https://gettingthingsdone.com/getting-things-done-the-art-of-stress-free-productivity/)

JamesBot frees your mind of having to remember things you need to do. It's,

- text-based

- easy to learn

- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download it from here.
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 👍 

And it is **FREE!**

Features:

- [x] Managing tasks

- [ ] Managing deadlines (coming soon)

- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

JunHao247 committed:
Level-1

JunHao247 committed:
Level-1

JunHao247 committed:
Level-1

JunHao247 committed:
Level-1

JunHao247 committed:
Level-2

JunHao247 committed:
Level-3

JunHao247 committed:
no message

JunHao247 committed:
Level-4

JunHao247 committed:
A-TextUiTesting

JunHao247 committed:
Level-5

JunHao247 committed:
Level-5

JunHao247 committed:
Level-6

JunHao247 committed:
Level-6

JunHao247 committed:
Level-7

JunHao247 committed:
Update Deadline class

JunHao247 committed:
no message

JunHao247 committed:
Date and Time formatting

JunHao247 committed:
Date and Time formatting

JunHao247 committed:
Merge branch 'branch-Level-8'

Resolved merge conflict

JunHao247 committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

JunHao247 committed:
Added Gradle Support

JunHao247 committed:
Gradle fix

JunHao247 committed:
Update testing classes for tasks

JunHao247 committed:
A-JUnit

JunHao247 committed:
Update classes with OOP principles
Include Parser, Command, Storage, TaskList classes as well as the additional command classes

JunHao247 committed:
A-OOP

JunHao247 committed:
Resolve merge conflicts

JunHao247 committed:
Update ‘james’, ‘task’, ‘parser’, ‘command’ packages

JunHao247 committed:
A-Packages

JunHao247 committed:
Update class names for command package to PascalCase

JunHao247 committed:
Update imported classes to be explicit

JunHao247 committed:
Update method name to camelCase

JunHao247 committed:
Add JavaDoc and new package ui

JunHao247 committed:
Update with Main.java and new packages storage and ui

JunHao247 committed:
Add JavaDoc to most classes

JunHao247 committed:
Update JavaDoc comments for most command classes and parser class

JunHao247 committed:
Add equals method and FindCommand for Level-9

JunHao247 committed:
Merge branch 'A-CodingStandard'

* A-CodingStandard:
  Update with Main.java and new packages storage and ui
  Update method name to camelCase
  Update imported classes to be explicit
  Update class names for command package to PascalCase

JunHao247 committed:
Merge branch 'A-JavaDoc'

* A-JavaDoc:
  Update JavaDoc comments for most command classes and parser class
  Add JavaDoc to most classes
  Add JavaDoc and new package ui

# Conflicts:
#	src/main/java/Main.java
#	src/main/java/james/James.java
#	src/main/java/james/command/AddDeadlineCommand.java
#	src/main/java/james/command/AddEventCommand.java
#	src/main/java/james/command/AddListCommand.java
#	src/main/java/james/command/AddMarkCommand.java
#	src/main/java/james/command/AddToDoCommand.java
#	src/main/java/james/command/AddUnmarkCommand.java
#	src/main/java/james/command/DeleteCommand.java
#	src/main/java/james/storage/Storage.java

JunHao247 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add equals method and FindCommand for Level-9

# Conflicts:
#	data/james.txt
#	src/main/java/james/parser/Parser.java
#	src/main/java/james/task/Task.java

JunHao247 committed:
Update CodingStandard

JunHao247 committed:
Update README according to GFMD standards

BigheartSmallbrain commented on src/main/java/james/ui/UI.java:
```diff
@@ -0,0 +1,116 @@
+ * The user interface class.
+ * Handles all user's input and output.
+ */
+public class UI {
```
I think it would be better not to uppercase abbreviation for naming convention. What do yout think?

JunHao247 replied:
According to the coding standard it should be Ui instead, thank you for spotting

BigheartSmallbrain commented on src/main/java/james/parser/Parser.java:
```diff
@@ -0,0 +1,101 @@
+
+
+    public Command parseCommand(String input) throws JamesException {
+                if (input.equals("list")) {
```
Should these block of if-else statements have 4 space or 8 space indentation?

JunHao247 replied:
Not sure what went wrong with my IDE, was 4 spaces before, will make the change, thank you!

BigheartSmallbrain commented on src/main/java/james/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+            while (sc.hasNextLine()) {
+                String[] task = sc.nextLine().split(" \\| ");
+                switch (task[0]) {
+                    case "T":
```
Should there be an indentation for case block of codes?

JunHao247 replied:
edited to have no indentation for cases

BigheartSmallbrain reviewed (commented):
LGTM overall, just some nits to fix. The codes are well organised and can be understood. Demonstrates most of the coding standards.

jeraldkiew commented on src/main/java/james/task/Task.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
boolean naming follows best practices 👍🏻

JunHao247 replied:
thank you

jeraldkiew commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1 @@
+todo borrow book
```
test input could be more comprehensive, perhaps at least touch on each of your main features :D

JunHao247 replied:
At this moment in time I did not manage to push my test inputs with deadline and events unfortunately, but thank you for the comment

jeraldkiew commented on src/main/java/james/command/AddListCommand.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * The command to list all tasks in the task list.
+ */
+public class AddListCommand extends Command {
```
ListTasksCommand might be a more appropriate class name 👍🏻

JunHao247 replied:
That is appropriate for a change thank you

jeraldkiew reviewed (commented):
Overall I think your code is well done! There is very good compartmentalisation to organise code for their respective features and I've learnt a couple of things for myself as well 👍🏻

JunHao247 committed:
Update coding standards according to peer reviewers' comments, change UI to Ui, change indentation for cases in switch statement to the proper indentation, change AddListCommand to ListTasksCommand

JunHao247 committed:
Update GUI for Level-10

JunHao247 committed:
Update event to arguments that work

JunHao247 committed:
Update event, deadline class to work with the required inputs

JunHao247 committed:
Update unnecessary methods

JunHao247 committed:
Update file that saves tasks

JunHao247 committed:
There are no checks done to prevent programmer's mistakes. These include
FXML components being improperly configured and the storage file not
existing.

Add assertions to check for these mistakes and print out an error
message.

Using assertions is preferable over exceptions as assertions indicate
that the programmer has made a mistake in their code rather than a
mistake made by the user.

JunHao247 committed:
The code could have been more readable.
There were also some assumptions made about the user's input such as
the user's commands containing spaces. These assumptions can be
dangerous as runtime errors can occur.

Add respective classes AddToDoCommand, AddDeadlineCommand,
AddEventCommand for different AddCommand.
Include constants for all the classes to make
code exception more readable.

JunHao247 committed:
Update test files and add assert methods to ensure no programmer's
mistake are present

JunHao247 committed:
Merge pull request #2 from JunHao247/branch-A-Assertions

Add assertions

JunHao247 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
Update test files and add assert methods to ensure no programmer's
mistake are present
There are no checks done to prevent programmer's mistakes.
These include FXML components being improperly configured and
the storage file not existing.

Update file that saves tasks

# Conflicts:
#	data/james.txt
#	src/main/java/james/command/AddCommand.java
#	src/main/java/james/command/AddMarkCommand.java
#	src/main/java/james/task/TaskList.java

JunHao247 committed:
Merge pull request #3 from JunHao247/branch-A-CodeQuality

Update according to code quality standard

JunHao247 committed:
Include C-Help as an extension category, it provides in-App guidance to
users. Add a command "help" to access help page.

JunHao247 committed:
Update according to checkstyle.xml, fix all the checkstyle errors and
violations

JunHao247 committed:
Update names of variables and constants to be clearer.

Change names of constants in Command classes.

This is to make the code to have better readability

JunHao247 committed:
Update Command classes names of variables and constants to have better
readability
Change font for nicer GUI

JunHao247 committed:
Update names of variables and constants to improve readability,
Update Javadoc to appropriately state the functionality
Update reuse comments
Rectify mistake in exception of Mark and Unmark command

JunHao247 committed:
Update JavaFX to include background photo and resize window

JunHao247 committed:
Merge pull request #4 from JunHao247/branch-A-BetterGui

Branch- Better Gui

JunHao247 committed:
Update User Guide based on GFMD to the product website

JunHao247 committed:
Update README.md

lukkwen committed:
Update ReadMe userguide

JunHao247 committed:
Update README.md

JunHao247 committed:
Update README.md

JunHao247 committed:
Update README.md

JunHao247 committed:
Update javadoc comments and javafx to fix window sizing and scaling of
background picture
